WalletWasabi/Wallets/P2pBlockProvider.cs

```diff
@@ -127,14 +127,14 @@ public async Task<Block> GetBlockAsync(uint256 hash, CancellationToken cancellat
 
 							await NodeTimeoutsAsync(true).ConfigureAwait(false);
 
-							node.DisconnectAsync("Block download took too long.");
+							DisconnectNode(node, "Block download took too long."); // it could be a slow connection and not a misbehaving node
```

Comment:
We log that we disconnected the node above, but this might be not true.

Reply:
Thank you. Done.